src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -263,28 +266,39 @@ struct PhotonCreator
         const float_X gamma = Gamma<>()(mom, mass);
         const float3_X vel = mom / (gamma * mass); // low accuracy?
 
-        const float3_X lorentz = fieldE + math::cross(vel, fieldB);
-        const float_X lorentz2 = math::dot(lorentz, lorentz);
+        const float3_X lorentzForceOverCharge = fieldE + math::cross(vel, fieldB);
+        const float_X lorentzForceOverCharge2 = math::dot(lorentzForceOverCharge, lorentzForceOverCharge);
         const float_X fieldE_long = math::dot(mom_norm, fieldE);
 
-        // effective magnetic strength
-        const float_X H_eff = math::sqrt(lorentz2 - fieldE_long*fieldE_long);
+        // effective magnetic strength (in cgs)
+        const float_X H_eff = math::sqrt(lorentzForceOverCharge2 - fieldE_long*fieldE_long);
 
         const float_X charge = math::abs(frame::getCharge<FrameType>());
 
         const float_X c = SPEED_OF_LIGHT;
-        // Schwinger limit, unit: V/m
+        // Schwinger limit, unit: V/m (in cgs)
         const float_X E_S = mass*mass * c*c*c / (charge * HBAR);
         // quantum-nonlinearity parameter
         const float_X chi = gamma * H_eff / E_S;
 
-        const float_X z = this->randomGen();
+        const float_X deltaScaled = this->randomGen();
 
-        const float_X x = emission_prob_scaled(z, chi, gamma);
+        const float_X x = emission_prob_scaled(deltaScaled, chi, gamma);
+
+        // raise a warning if the emission probability is too high.
+        if(picLog::log_level & picLog::CRITICAL::lvl)
+        {
+            if(x > float_X(SINGLE_EMISSION_PROB_LIMIT))
+            {
+                printf("[SynchrotronPhotons] warning: emission probability is too high: p = %g, at delta = %g, chi = %g, gamma = %g\n",
```

Comment:
as discussed with @psychocoderHPC *offline*:
since this runs on the device, `log` will not work (it uses `std::cout`) - thus using `printf` is the only option :+1: